### PR TITLE
Replace deprecated 'NewGomegaWithT' with 'NewWithT'

### PR DESCRIPTION
**Pull request by frbimo**
Replace  'NewGomegaWithT' in favor of  'NewWithT' + fix some typo 

tested:
make test

### CODE REVIEW

**Comment by adrianludwin** [comment]
/approve
/assign @yiqigao217

Yiqi can you ptal?

File: incubator/hnc/internal/object/object_test.go [hunk 1]
```diff
@@ -1,9 +1,10 @@
 package object
 
 import (
-	. "github.com/onsi/gomega"
 	"testing"
 
+	. "github.com/onsi/gomega"
+
```
Comment (yiqigao217) [inline incubator/hnc/internal/object/object_test.go:7]: Ah right, it should be in the 2nd section. Could you please remove the empty line though?

**Review by yiqigao217 — COMMENTED** [review]
lgtm after 1 nit, thanks!

File: incubator/hnc/internal/object/object_test.go (hunk 1 above)
Comment (frbimo) [reply incubator/hnc/internal/object/object_test.go]: done

**Comment by adrianludwin** [comment]
/ok-to-test

@yiqigao217 lgty?

**Review by yiqigao217 — COMMENTED** [review]
/lgtm

**Pull request merged**